src/corecel/math/TridiagonalSolver.hh

```diff
@@ -0,0 +1,48 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/math/TridiagonalSolver.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <vector>
+
+#include "corecel/Assert.hh"
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "corecel/cont/Span.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Solve a tridiagonal system of equations using the Thomas algorithm.
+ *
+ * This is a simplified form of Gaussian elimination that can solve a
+ * tridiagonal system \f$ a_i x_{i - 1} + b_i x_i + c_i x_{i + 1} = d_i \f$
+ * with \f$ n \f$ unknowns where \f$ a_1 = 0 \f$ and \f$ c_n = 0 \f$ in O(n)
+ * time.
+ */
```

Comment:
Even thought it might be slightly more efficient to smash the LHS and RHS together, I think this class would be better written as "apply T^{-1}"; where T is the tridiagonal. Since I guess you want to "return" into the middle of an existing array I think we can make this class behave like `x = A \ b` in matlab, but returning by reference: `void  operator()(Span<real_type const> rhs, Span<real_type> x);` In theory if we need to do this to multiple `b` simultaneously we could overload for better efficiency.